chapter04/chapter04.adoc

```diff
@@ -0,0 +1,317 @@
+
+This specification streamlines the creation of OpenAPI documentation for RESTful services using a unified approach. It generates comprehensive metadata about services, ensuring interoperability across diverse platforms and tools. The OpenAPI Specification fuels a rich ecosystem of tools that automate and support:
+
+* *API Documentation Generation*: Intuitive interactive documentation portals emerge directly from the specification.
+* *Client SDK Creation*: Client libraries in various languages can be automatically generated.
+* *API Testing*: Testing frameworks can leverage the specification to design robust tests.
+* *API Mocking*: Simplifies mocking APIs for testing and development purposes.
+
+=== Generating OpenAPI documents
+
+There are multiple ways in which you can generate OpenAPI documents. The most common way is to use annotations. This only requires augmenting your Jakarta Restful Web Services annotations with OpenAPI annotations. 
+
+You can also use JSON or YAML files. You can start with a predefined model and manually update the entire OpenAPI document tree of your web services. However, the annotation-based approach is recommended as it is more maintainable and easier to understand. Finally, you can filter out the resources you do not want to document using configuration.
+
+== Using MicroProfile Open API in your project
+
+To document Jakarta RESTful Web Services using MicroProfile OpenAPI, we need to annotate the resource classes and methods with the OpenAPI annotation. 
+
+To use MicroProfile OpenAPI in your project, you need to add the following maven coordinates to your project:
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.eclipse.microprofile.openapi</groupId>
+  <artifactId>microprofile-openapi-api</artifactId>
+  <version>3.1.1</version>
+</dependency>
+----
+
+Below is an illustrative example of how you might annotate a method in the `ProductResource` class to achieve this documentation using MicroProfile OpenAPI annotations:
+
+[source, java]
+----
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import java.util.List;
+
+@Path("/products")
+public class ProductResource {
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "List all products", description = "Retrieves a list of all available products")
+    @APIResponses(value = {
```

Comment:
`@APIResponse` is repeatable, so you may omit `@APIResponses` if that helps it be more readable.

Reply:
Keeping it within container annotations to demonstrate it usage and better organization.